### PR TITLE
lapack/gonum: add Dlansb

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

### CODE REVIEW

File: lapack/gonum/dlansb.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	case lapack.Frobenius:
+		scale := 0.0
+		ssq := 1.0
+		if uplo == blas.Upper {
+			if kd > 0 {
+				// Sum off-diagonals.
+				for i := 0; i < n-1; i++ {
+					ilen := min(n-i-1, kd)
+					rowscale, rowssq := impl.Dlassq(ilen, ab[i*ldab+1:], 1, 0, 1)
+					scale, ssq = impl.Dcombssq(scale, ssq, rowscale, rowssq)
```
Comment (kortschak) [inline lapack/gonum/dlansb.go]: Where is the `Dcombssq` call from? I don't see that in LAPACK 3.8.0.

File: lapack/testlapack/dcombssq.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			scale2 = 0
+		case 2:
+			scale1, scale2 = 0, 0
+		}
+
+		// Compute scale and ssq such that
+		//  scale^2 * ssq := scale1^2 * ssq1 + scale2^2 * ssq2
+		scale, ssq := impl.Dcombssq(scale1, ssq1, scale2, ssq2)
+
+		// Compute the expected result in a non-sofisticated way and
```
Comment (kortschak) [inline lapack/testlapack/dcombssq.go]: sophisticated

File: lapack/testlapack/dlansb.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+	"gonum.org/v1/gonum/blas"
```
Comment (kortschak) [inline lapack/testlapack/dlansb.go]: Blank line between us and them.

File: lapack/testlapack/dlassq.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+			// Allocate a slice of minimum length and fill it with
+			// random numbers.
+			x := make([]float64, max(0, 1+(n-1)*incx))
+			for i := range x {
+				x[i] = rnd.Float64()
+			}
+
+			// Fill the referenced elements of x and compute the
+			// expected result in a non-sofisticated way.
```
Comment (kortschak) [inline lapack/testlapack/dlassq.go]: sophisticated

File: lapack/gonum/dlansb.go (hunk 1 above)
Comment (vladimir-ch) [reply lapack/gonum/dlansb.go]: That's a relatively recent addition: https://github.com/Reference-LAPACK/lapack/pull/290

File: lapack/testlapack/dcombssq.go (hunk 2 above)
Comment (vladimir-ch) [reply lapack/testlapack/dcombssq.go]: Oh.

File: lapack/testlapack/dlansb.go (hunk 3 above)
Comment (vladimir-ch) [reply lapack/testlapack/dlansb.go]: When the machine does 99% of what you need ...

Done.

File: lapack/testlapack/dlassq.go (hunk 4 above)
Comment (vladimir-ch) [reply lapack/testlapack/dlassq.go]: Oh.

**Comment by vladimir-ch** [comment]
PTAL

File: lapack/gonum/dlansb.go (hunk 1 above)
Comment (kortschak) [reply lapack/gonum/dlansb.go]: Thanks

**Pull request merged**